google/cloud/aiplatform/metadata/artifact.py

```diff
@@ -93,8 +226,257 @@ def _list_resources(
+        Returns:
+            Artifact: Artifact with given uri.
+        Raises:
+            ValueError: If no Artifact exists with the provided uri.
+
+        """
+
+        matched_artifacts = cls.list(
+            filter=f'uri = "{uri}"',
+            metadata_store_id=metadata_store_id,
+            project=project,
+            location=location,
+            credentials=credentials,
+        )
+
+        if not matched_artifacts:
+            raise ValueError(
+                f"No artifact with uri {uri} is in the `{metadata_store_id}` MetadataStore."
+            )
+
+        if len(matched_artifacts) > 1:
+            resource_names = "\n".join(a.resource_name for a in matched_artifacts)
+            _LOGGER.warn(
+                f"Mutiple artifacts with uri {uri} were found: {resource_names}"
+            )
+            _LOGGER.warn(f"Returning {matched_artifacts[0].resource_name}")
+
+        return matched_artifacts[0]
+
+    @property
+    def lineage_console_uri(self) -> str:
+        """Cloud console uri to view this Artifact Lineage."""
+        metadata_store = self._parse_resource_name(self.resource_name)["metadata_store"]
+        return f"https://console.cloud.google.com/vertex-ai/locations/{self.location}/metadata-stores/{metadata_store}/artifacts/{self.name}?project={self.project}"
+
+    def __repr__(self) -> str:
+        if self._gca_resource:
+            return f"{object.__repr__(self)} \nresource name: {self.resource_name}\nuri: {self.uri}\nschema_title:{self.gca_resource.schema_title}"
+
+        return base.FutureManager.__repr__(self)
+
+
+class _VertexResourceArtifactResolver:
+
+    _resource_to_artifact_type = {models.Model: "google.VertexModel"}
+
+    @classmethod
+    def supports_metadata(cls, resource: base.VertexAiResourceNoun) -> bool:
+        """Returns True if Vertex resource is supported in Vertex Metadata otherwise False.
+
+        Args:
+            resource (base.VertexAiResourceNoun):
+                Requried. Instance of Vertex AI Resource.
+        Returns:
+            True if Vertex resource is supported in Vertex Metadata otherwise False.
+        """
+        return type(resource) in cls._resource_to_artifact_type
+
+    @classmethod
+    def validate_resource_supports_metadata(cls, resource: base.VertexAiResourceNoun):
+        """Validates Vertex resource is supported in Vertex Metadata.
+
+        Args:
+            resource (base.VertexAiResourceNoun):
+                Required. Instance of Vertex AI Resource.
+        Raises:
+            ValueError: If Vertex AI Resource is not support in Vertex Metadata.
+        """
+        if not cls.supports_metadata(resource):
+            raise ValueError(
+                f"Vertex {type(resource)} is not supported in Vertex Metadata."
+                f"Only {list(cls._resource_to_artifact_type.keys())} are supported"
+            )
+
+    @classmethod
+    def resolve_vertex_resource(
+        cls, resource: Union[models.Model]
```

Comment:
It can take only 1: https://docs.python.org/3/library/typing.html#typing.Union

This is to represent the intention that additional Vertex resources will be supported.